sonar-ruby-plugin/src/main/java/org/sonarsource/ruby/converter/RubyConverter.java

```diff
@@ -0,0 +1,131 @@
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.ruby.converter;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import org.jruby.Ruby;
+import org.jruby.RubyRuntimeAdapter;
+import org.jruby.exceptions.NoMethodError;
+import org.jruby.exceptions.StandardError;
+import org.jruby.javasupport.JavaEmbedUtils;
+import org.jruby.runtime.builtin.IRubyObject;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonarsource.ruby.converter.adapter.RangeAdapter;
+import org.sonarsource.slang.api.ASTConverter;
+import org.sonarsource.slang.api.TextPointer;
+import org.sonarsource.slang.api.Tree;
+import org.sonarsource.slang.impl.TopLevelTreeImpl;
+import org.sonarsource.slang.plugin.ParseException;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Collections.emptyList;
+
+public class RubyConverter implements ASTConverter {
+
+  private static final Logger LOG = Loggers.get(RubyConverter.class);
+
+  private static final Path SETUP_SCRIPT_PATH = Paths.get("/whitequark_parser_init.rb");
+  private static final Path AST_RUBYGEM_PATH = Paths.get("/ast-2.4.0", "lib");
+  private static final Path PARSER_RUBYGEM_PATH = Paths.get("/parser-2.5.1.2", "lib");
+  private static final RubyRuntimeAdapter rubyRuntimeAdapter = JavaEmbedUtils.newRuntimeAdapter();
+
+  private final Ruby runtime;
+
+  public RubyConverter() {
+    try {
+      runtime = initializeRubyRuntime();
+    } catch (URISyntaxException | IOException e) {
+      throw new IllegalStateException("Failed to initialized ruby runtime", e);
+    }
+  }
+
+  public void terminate() {
+    // Shutdown and terminate ruby instance
+    if (runtime != null) {
+      JavaEmbedUtils.terminate(runtime);
+    }
+  }
+
+  @Override
+  public Tree parse(String content) {
+    try {
+      return parseContent(content);
+    } catch (StandardError e) {
+      throw new ParseException(e.getMessage(), getErrorLocation(e));
+    }
+  }
+
+  private TextPointer getErrorLocation(StandardError e) {
```

Comment:
It was actually not a FP. Because of the `Nonnull` default behaviour of methods, and `invokeMethod` was not annotated as `Nullable`.
Can you check it does not raise the issue on your side anymore too?

Reply:
it's ok on my side